mne/io/base.py

```diff
@@ -2500,25 +2500,35 @@ def _write_raw_buffer(fid, buf, cals, fmt):
 
 def _check_raw_compatibility(raw):
     """Ensure all instances of Raw have compatible parameters."""
-    for ri in range(1, len(raw)):
-        if not isinstance(raw[ri], type(raw[0])):
-            raise ValueError(f'raw[{ri}] type must match')
-        for key in ('nchan', 'bads', 'sfreq'):
-            a, b = raw[ri].info[key], raw[0].info[key]
+    for i in range(1, len(raw)):
```

Comment:
please put back as `ri`. We try to avoid bare `i` (hard to read, get confused with `1j`, etc.) and `ri` is even a tiny bit more informative as it makes it clear it's the "raw iterator" rather than just "iterator".